load_testing/README.md

```diff
@@ -0,0 +1,157 @@
+
+## Saved to file
+
+Results are saved to file at 4 stages:
+
+1. During each test run
+2. End of each test run
+3. End of each experiment's set of test runs
+4. End of all experiments
+
+
+Results from each individual load-testing experiment are also saved under a corresponding `experiment_name/` subfolder. The following subfolders are then created inside:
+
+1. `raw/`: While each test is running, raw stats and error info are flushed to disk every 10 seconds by Locust into a further subfolder titled `[users]_user_[locustfile]/`, corresponding to the test parameters. Files saved by Locust:
+
+    ```console
+    test_exceptions.csv
+    test_failures.csv
+    test_stats.csv
+    test_stats_history.csv
+    ```
+
+2. `html_reports/`: At the end of each test, an HTML report made by Locust is saved here. Each report includes final stats and load-test progression plots.
+
+### Example output folder structure
+
+```console
+📂outputs
+┣ all_experiments_endoftest_results_summary.csv
+┣ 📂staging_constant_multi
+┃ ┣ 📂html_reports
+┃ ┃ ┣ 10_user_same_msgs_report.html
+┃ ┃ ┗ ...
+┃ ┣ 📂raw
+┃ ┃ ┣ 10_user_same_msgs_report
+┃ ┃ ┃ ┣ test_exceptions.csv
+┃ ┃ ┃ ┣ test_failures.csv
+┃ ┃ ┃ ┣ test_stats.csv
+┃ ┃ ┃ ┗ test_stats_history.csv
+┃ ┃ ┣ 📂100_user_val_msgs
+┃ ┃ ┗ ┗ ...
+┃ ┣ 📂staging_ramped
+┗ ┗ ┗ ...
+```
+
+# Config and experiment types
+
+## Config file format
+
+The config file used here is `.json` as opposed to locust's expected `.ini` style. This allows us to write defintions for multiple experiments without having to make a different config file for each, and is parsed automatically by the main script.
+
+Each key-value pair here is a load-test experiment, where the key is the experiment name and the value is a dictionary of parameters expected by `locust`.
+
+Each experiment entry reflects the core elements of a [standard locust config file](https://docs.locust.io/en/stable/configuration.html) but with the key difference that parameters are given as lists as opposed to single values. This allows us to run multiple test configurations per experiment, namely to use different locustfiles (type of requests sent) and numbers of users.
+
+## Experiment types
+
+This submodule is designed to be able to run two types of experiments: constant and ramped load-tests. As many experiments can be added to the config file as necessary - the script will execute one after the other.
+
+### _Constant load-tests_
+
+Constant load-tests initiate the max number of users from the start and sustain the load for the given run-time.
+
+When multiple request types (via different locustfiles) and numbers of users are specificed, the constant load-tests experiment outputs a grid of results that allows us to compare response times and requests/sec under stable load conditions.
+
+An example of a config entry for multiple constant load-tests performed on the `STAGING_URL` is given below:
```

Comment:
update mentions of STAGING_URL? We don't use that terminology anymore (at least not yet)